crates/jarscanner/src/lib.rs

```diff
@@ -99,10 +131,11 @@ pub fn process_input(
 pub fn emit_result(
     target_descriptor: &ExtractedData,
     output_path: &PathBuf,
-) -> Result<(), Box<dyn Error>> {
+) -> Result<(), JarscannerError> {
     let json = serde_json::to_string_pretty(target_descriptor)?;
-    let mut file = File::create(output_path)?;
-    file.write_all(json.as_bytes())?;
+    let file = File::create(output_path)?;
+    let mut writer = BufWriter::new(file);
```

Comment:
this defaults to 8kb: https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.new

maybe instead use `with_capacity` so we can see how big and maybe use something bigger like `32k` or something?